source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRTableRow.mm

```diff
@@ -0,0 +1,154 @@
+                    cellDefinition.horizontalAlignment = (ACRHorizontalAlignment)row->GetHorizontalCellContentAlignment().value_or(HorizontalAlignment::Left);
+                    cellDefinition.verticalAlignment = (ACRVerticalAlignment)row->GetVerticalCellContentAlignment().value_or(VerticalContentAlignment::Top);
+                    cellView = [[ACRTableCellView alloc] init:[[ACOBaseCardElement alloc] initWithBaseCardElement:cell]
+                                               cellDefinition:cellDefinition
+                                                     rootView:rootView
+                                                       inputs:inputs
+                                                   hostConfig:acoConfig];
+                } else {
+                    // filler view for empty cells
+                    cellView = [[ACRTableCellView alloc] init];
+                }
+                cellView.translatesAutoresizingMaskIntoConstraints = NO;
+
+                if (cellView) {
+                    [self addSubview:cellView];
+                    if (ithColumnDefinition.isPixelWidth) {
+                        [cellView.widthAnchor constraintEqualToConstant:ithColumnDefinition.numeric].active = YES;
+                    } else if (!ithColumnDefinition.isPixelWidth) {
+                        [cellView.widthAnchor constraintEqualToAnchor:self.widthAnchor
+                                                           multiplier:ithColumnDefinition.numeric
+                                                             constant:-ithColumnDefinition.totalPixelWidth]
+                            .active = YES;
+                    }
+
+                    [self.heightAnchor constraintGreaterThanOrEqualToAnchor:cellView.heightAnchor].active = YES;
+                    NSLayoutConstraint *heightConstraint = [cellView.heightAnchor constraintGreaterThanOrEqualToConstant:0.0];
+                    heightConstraint.priority = UILayoutPriorityDefaultLow;
+                    heightConstraint.active = YES;
+
+                    [cellView.topAnchor constraintEqualToAnchor:self.topAnchor].active = YES;
+                    if (prevView) {
+                        [cellView.leadingAnchor constraintEqualToAnchor:trailingAnchor constant:spacing].active = YES;
+                    } else {
+                        [cellView.leadingAnchor constraintEqualToAnchor:trailingAnchor].active = YES;
+                    }
+                    trailingAnchor = cellView.trailingAnchor;
+                    prevView = cellView;
+                }
+            }
+        }
+    }
+    return self;
+}
+
+- (void)addArrangedSubview:(UIView *)view
+{
+    [self addSubview:view];
+}
+
+- (CGSize)intrinsicContentSize
+{
+    CGSize size = CGSizeZero;
+    for (UIView *subview in self.subviews) {
+        if (!subview.isHidden) {
+            CGSize intrinsicContentSize = [subview intrinsicContentSize];
+            size.width += intrinsicContentSize.width;
```

Comment:
does this calculation take into consideration the spacing?

Reply:
thanks, good catch

Reply:
I'll make the change in 2/3 pr.